components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/core/mapper/XSLTMappingResource.java

```diff
@@ -0,0 +1,172 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.wso2.carbon.mediator.datamapper.engine.utils.DataMapperEngineConstants
+        .EMPTY_STRING;
+import static org.wso2.carbon.mediator.datamapper.engine.utils.DataMapperEngineConstants
+        .FIRST_ELEMENT_OF_THE_INPUT;
+import static org.wso2.carbon.mediator.datamapper.engine.utils.DataMapperEngineConstants
+        .NOT_XSLT_COMPATIBLE;
+import static org.wso2.carbon.mediator.datamapper.engine.utils.DataMapperEngineConstants
+        .PARAMETER_FILE_ROOT;
+import static org.wso2.carbon.mediator.datamapper.engine.utils.DataMapperEngineConstants
+        .PROPERTY_SEPERATOR;
+import static org.wso2.carbon.mediator.datamapper.engine.utils.DataMapperEngineConstants
+        .RUN_TIME_PROPERTIES;
+import static org.wso2.carbon.mediator.datamapper.engine.utils.DataMapperEngineConstants
+        .XSLT_COMPATIBLE_DEFAULT;
+
+public class XSLTMappingResource {
+
+    private final Map<String, String> runTimeProperties;
+    private String name;
+    private String content;
+    private boolean notXSLTCompatible;
+
+
+    public XSLTMappingResource(String content) throws SAXException,
+            IOException,
+            ParserConfigurationException {
+        this.content = content;
+        this.runTimeProperties = new HashMap<>();
+        Document document = getDocument();
+        notXSLTCompatible = processConfigurationDetails(document);
+        if (notXSLTCompatible) {
+            this.content = null;
+        }
+    }
+
+    /**
+     * Create a input stream from the available content
+     *
+     * @return Input stream of the xslt stylesheet
+     */
+    InputStream getInputStream() {
+        return new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Create a input source from the available content
+     *
+     * @return Input source of the xslt stylesheet
+     */
+    private InputSource getInputSource() {
+        return new InputSource(new StringReader(content));
+    }
+
+    /**
+     * Creating a document to process the xslt stylesheet
+     *
+     * @return document of the xslt stylesheet
+     * @throws SAXException
+     * @throws IOException
+     * @throws ParserConfigurationException
+     */
+    private Document getDocument() throws SAXException, IOException,
+            ParserConfigurationException {
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+        return documentBuilder.parse(getInputSource());
+    }
+
+    /**
+     * Process configuration details included in the xslt stylesheet
+     *
+     * @param document xslt stylesheet as a document
+     * @return return whether the xslt transformation possible or not
+     */
+    private boolean processConfigurationDetails(Document document) {
+        boolean not_compatible = true;
```

Comment:
I suppose instead of not_compatible = true, we can use compatible = false. It'd be easy to interpret. WDYT?

Reply:
Yes, it would be convenient. I have changed it now.